controllers/urls.js

```diff
@@ -1,16 +1,51 @@
-const express = require('express')
-const Url = require('../models/Url')
+const express = require('express');
+const { validationResult } = require('express-validator');
+const statusCodes = require('http-status-code');
+const Url = require('../models/Url');
+const generateRandomString = require('../middleware/randomString');
 
-const createShortUrl = async(req, res) => {
-    const url = await Url.create({...req.body})
-    res.json(url)
-}
+const createShortUrl = async (req, res) => {
+  const { customName, originalUrl } = req.body;
+  let shortUrls;
 
-const customShortUrl = async(req, res)  => {
-    res.send('custom name for short url')
-}
+  try {
+    if (customName) {
+      const customNameWithDashes = customName.split(' ').join('-');
+      shortUrls = `https://shortit/${customNameWithDashes}`;
+      const existingCustomName = await Url.findOne({ customName: customNameWithDashes });
+      if (existingCustomName) {
+        return res.status(400).json({ errors: 'Custom name already exists' });
+      }
+    } else {
+      const randomString = generateRandomString(5);
+      shortUrls = `https://shortit/${randomString}`;
+    }
+
+    const existingRandomString = await Url.findOne({ shortUrl: shortUrls });
+    if (existingRandomString) {
+      return res.status(400).json({ error: 'This string already exists' });
+    }
+    const existingLongUrl = await Url.findOne({ originalUrl });
+    if (existingLongUrl) {
+      res.status(200).json({ shortUrl: shortUrls });
+    }
+    const url = await Url.create({
+      ...req.body,
+      shortUrl: shortUrls,
+      customName
+    });
+    res.status(200).json(url);
```

Comment:
When I hit the post route repeatedly, I get this error. Do you get this?

<img width="1635" alt="Screenshot 2023-10-14 at 12 53 28" src="https://github.com/PraiseTy/Short-It/assets/48496855/7c3bb956-c7a1-4dc5-804c-b823601b8ec4">

Reply:
I have gotten this error but didn't know what caused it. I did some research and its  because the code is trying to send multiple response at once and since you highlighted lines 29 to 37, then i have to assume the error was there and I added a return before the res.status to help fix that and it's done on 5fb15b1. Also please don't mind the typos in the commit name, mistake on my part

Reply:
Yes. Adding return will fix it.